tests/utils/utils.go

```diff
@@ -568,6 +571,35 @@ func WaitMined(ctx context.Context, b bind.DeployBackend, txHash common.Hash) (*
 		case <-queryTicker.C:
 		}
 	}
+
+	// Check that the block height endpoint returns a block height as high as the block number that the transaction was
+	// included in. This is to workaround the issue where multiple nodes behind a public RPC endpoint see
+	// transactions/blocks at different points in time. Ideally, all nodes in the network should have seen this block
+	// and transaction before returning from waitMined. The block height endpoint of public RPC endpoints is
+	// configured to return the lowest value currently returned by any node behind the load balancer, so waiting for
+	// it to be at least as high as the block height specified in the receipt should provide a relatively strong
+	// indication that the transaction has been seen widely throughout the network.
```

Comment:
So just double-checking my understanding: The loop above waits for the transaction to be mined, and be available on the node we happen to be hitting. Then the loop below here, will loop until that block has been seen by all nodes behind the load-balancer.

Reply:
Yep that's correct. I'm not sure how strong a guarantee it is that _all_ nodes have seen the block height returned by the load balancer, but it is definitely a good sanity check that in my testing caused additional waiting.